proto/connect/datastore_service/v1alpha1/datastore_service.proto

```diff
@@ -0,0 +1,95 @@
+// Copyright 2025 Cofide Limited.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package proto.connect.datastore_service.v1alpha1;
+
+import "proto/connect/datastore_service/v1alpha1/datastore.proto";
+
+option go_package = "github.com/cofide/cofide-api-sdk/gen/go/proto/connect/datastore_service/v1alpha1";
+
+// DataStoreService defines the gRPC service for the SPIRE server DataStore interface
+service DataStoreService {
+  // Node operations
+  rpc CountAttestedNodes(CountAttestedNodesRequest) returns (CountAttestedNodesResponse);
+  rpc CreateAttestedNode(CreateAttestedNodeRequest) returns (CreateAttestedNodeResponse);
+  rpc DeleteAttestedNode(DeleteAttestedNodeRequest) returns (DeleteAttestedNodeResponse);
+  rpc FetchAttestedNode(FetchAttestedNodeRequest) returns (FetchAttestedNodeResponse);
+  rpc UpdateAttestedNode(UpdateAttestedNodeRequest) returns (UpdateAttestedNodeResponse);
+  rpc ListAttestedNodes(ListAttestedNodesRequest) returns (ListAttestedNodesResponse);
+  // Node Selector operations
+  rpc GetNodeSelectors(GetNodeSelectorsRequest) returns (GetNodeSelectorsResponse);
+  rpc SetNodeSelectors(SetNodeSelectorsRequest) returns (SetNodeSelectorsResponse);
+  rpc ListNodeSelectors(ListNodeSelectorsRequest) returns (ListNodeSelectorsResponse);
+}
+
+message ListNodeSelectorsRequest {}
```

Comment:
Should we include `valid_at` field here?

Reply:
Yeah, I called this out in PR blurb - there are no filters yet, and I've raised an issue to cover this: https://github.com/cofide/cofide-api-sdk/issues/70

Reply:
done